pandas/indexes/base.py

```diff
@@ -86,6 +86,12 @@ def _new_Index(cls, d):
     """ This is called upon unpickling, rather than the default which doesn't
     have arguments and breaks __new__
     """
+    # required for backward compat, because PI can't be instantiated with
+    # ordinals through __new__ GH #13277
+    if issubclass(cls, ABCPeriodIndex):
+        if d['data'].dtype == 'int64':
+            values = d.pop('data')
```

Comment:
This needs an overall simplification, I'll add a todo. I think that proposal is better but not optimal.
Let's do that in another PR though, or I'll be doing another painful rebase in 2017!